pkg/state/state.go

```diff
@@ -70,22 +70,27 @@ LANGUAGE SQL
 STABLE;
 
 -- Get the name of the previous version of the schema, or NULL if there is none.
+-- This ignores previous versions for which no version schema exists, such as
+-- versions corresponding to inferred migrations.
```

Comment:
is this always the behavior? I wonder if this should be based on the setting

Reply:
It's equivalent to the old behaviour when the new setting is off. The only migrations that didn't create version schema were `inferred` migrations which were already filtered out by this function.

Reply:
The only use of this function is for cleaning up old version schema, for which this version of the function is fine. If we have a need to find the previous version, even if it doesn't have a version schema we can handle that then.